piptools/resolver.py

```diff
@@ -182,6 +184,12 @@ def _filter_out_unsafe_constraints(
 
 
 class LegacyResolver(BaseResolver):
+    """
+    Resolve a given set of constraints (a collection of
```

Comment:
It's a class, not a callable function. Classes represent static things/data. They don't do anything on their own. This is why, unlike functions/methods, their docstrings shouldn't start with a verb, because they don't perform actions, they just organize stuff. I usually recommend just stating what it is/represents in the docstring title line.